### PR TITLE
Repo add

### PR DESCRIPTION
**Pull request by andreabonanno**
**What type of PR is this?**

 /kind feature

**Any specific area of the project related to this PR?**

 /area cli

**What this PR does / why we need it**:
This PR adds `repo add` command to Falcoctl.
This command enables Falcoctl to keep an internal list of artifacts repositories, enabling future usages by other commands (see #103).

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new: repo add command
```

### CODE REVIEW

File: cmd/repo_add.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	}
+}
+
+func NewRepoAddCmd(options CommandOptions) *cobra.Command {
+	o := options.(*RepoAddOptions)
+
+	cmd := &cobra.Command{
+		Use:                   "add",
+		DisableFlagsInUseLine: true,
+		Short:                 "Adds an artifact repository to the falcotl cache",
```
Comment (leogr) [inline cmd/repo_add.go:52]:
```suggestion
		Short:                 "Adds an artifact repository to the falcoctl cache",
```

File: cmd/repo_add.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		Long:                  "Search a plugin inside the official Falco registry",
```
Comment (leogr) [inline cmd/repo_add.go:53]: The text does not match the action purpose. Maybe a leftover?

File: pkg/repo/types.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"time"
+)
+
+const (
+	timeformat = "2006-01-02 15:04:05"
+	writeperm  = 0777
```
Comment (leogr) [inline pkg/repo/types.go:12]: Why `0777` ?

File: pkg/repo/types.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+type RepoList struct {
+	Sources []RepoEntry `yaml:"sources"`
+}
+
+type RepoEntry struct {
+	Name string `yaml:"name"`
+	Url  string `yaml:"url"`
+	Date string `yaml:"updated"`
```
Comment (leogr) [inline pkg/repo/types.go:22]: It's not totally clear to me the purpose of storing the date here. Could you explain it, please?

**Review by leogr — COMMENTED** [review]
Overall, SGMT. Just found some minor issues (see below).

File: cmd/repo_add.go (hunk 2 above)
Comment (andreabonanno) [reply cmd/repo_add.go:53]: Yes indeed!

File: pkg/repo/types.go (hunk 3 above)
Comment (andreabonanno) [reply pkg/repo/types.go:12]: That's an error from my part, it should be `0666`. Thanks for the heads-up!

File: pkg/repo/types.go (hunk 4 above)
Comment (andreabonanno) [reply pkg/repo/types.go:22]: Falcoctl does not rely on `Date` for the logic of its commands: it's just extra info to print for the user, represents the last time this entry was fetched/updated. Do you think it's unnecessary? Thanks!

**Review by leogr — APPROVED** [review]

**Pull request merged**